File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/push.cc
#include "push.h"
using namespace std;

Push::Push(int c) : c(c){}

void Push::ejecutar(Programa* programa){
    programa->getPila().push(c);
    programa->inc();
}

string Push::mostrar(){
    return("push " + to_string(c));
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Push.java
public class Push extends Instruccion{
    private int c;

    //Constructor del parámetro.
    public Push(int c){
        this.c = c;
    }

    public void ejecutar(Programa programa){
        programa.getPila().push(c);
        programa.inc();
    }

    public String mostrar(){
        return "push " + c;
    }
}